tt_metal/impl/buffers/buffer.hpp

```diff
@@ -144,6 +144,8 @@ struct BufferPageMapping {
 inline namespace v0 {
 
 class Buffer final {
+    struct Private { explicit Private() = default; };
```

Comment:
The `Private` structure is used to limit access to the Buffer constructor, which can't be truly private, because of `std::make_shared`. This is the way recommended by CppReference documentation: https://en.cppreference.com/w/cpp/memory/enable_shared_from_this (see `Best` example)